### PR TITLE
Add inspector controls

### PR DESCRIPTION
**Pull request by paulwilde**
This is a rebased version of #1202 whilst also including #1198.

### CODE REVIEW

**Comment by aduth** [comment]
There are some failing tests here due to missing `key` prop and trailing spaces:

https://travis-ci.org/WordPress/gutenberg/jobs/244678746

As a quality-of-life tip, when rebasing, you needn't create a separate pull request, but rather can force push atop your existing branch (e.g. `git push -f origin add/inspector-controls`)

File: blocks/inspector-controls/checkbox-control/index.js [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+import BaseControl from './../base-control';
+import './style.scss';
+
+function CheckboxControl( { label, heading, checked, instanceId, onChange, ...props } ) {
+	const id = 'inspector-checkbox-control-' + instanceId;
+	const onChangeValue = ( event ) => onChange( event.target.value );
+
+	return (
+		<BaseControl label={ heading } id={ id }>
+			<label className="blocks-checkbox-control__label" htmlFor={ id }>
```
Comment (aduth) [inline blocks/inspector-controls/checkbox-control/index.js]: Missed this before. We shouldn't combine approaches of implicit and explicit labelling, between `htmlFor` and nesting input as a child of the label. From previous conversations with @afercia , `htmlFor` should be our preferred approach as it is better supported by screen readers.

https://github.com/WordPress/gutenberg/pull/527#discussion_r113741025

**Review by aduth — CHANGES_REQUESTED** [review]

**Review by aduth — APPROVED** [review]

**Comment by aduth** [comment]
This looks good to me 👍

File: blocks/inspector-controls/toggle-control/index.js [hunk 2]
```diff
@@ -0,0 +1,23 @@
+/**
+ * WordPress dependencies
+ */
+import { withInstanceId } from 'components';
+import Toggle from 'components/form-toggle';
```
Comment (aduth) [inline blocks/inspector-controls/toggle-control/index.js]: Minor: We should import from the top-level, i.e. `import { withInstanceId, Toggle } from 'components';`

File: blocks/inspector-controls/select-control/index.js [hunk 3]
```diff
@@ -0,0 +1,39 @@
+/**
+ * WordPress dependencies
+ */
+import { withInstanceId } from 'components';
+import { isEmpty } from 'lodash';
```
Comment (aduth) [inline blocks/inspector-controls/select-control/index.js]: `lodash` is an external dependency, not a WordPress dependency.

File: blocks/inspector-controls/radio-control/index.js [hunk 4]
```diff
@@ -0,0 +1,42 @@
+/**
+ * WordPress dependencies
+ */
+import { withInstanceId } from 'components';
+import { isEmpty } from 'lodash';
```
Comment (aduth) [inline blocks/inspector-controls/radio-control/index.js]: Same note about Lodash from previous.

File: blocks/inspector-controls/toggle-control/index.js [hunk 5]
```diff
@@ -0,0 +1,22 @@
+/**
+ * WordPress dependencies
+ */
+import { withInstanceId, Toggle } from 'components';
```
Comment (aduth) [inline blocks/inspector-controls/toggle-control/index.js]: Oof, while this is preferred, we're also not currently exporting `Toggle` from `components/index.js` as we should be.

**Comment by youknowriad** [comment]
I took the liberty to update this to fix the ToggleControl issue. I'm going to merge this to unblock #1191

Thanks awesome work here.

**Pull request merged**